solarwinds/stackstate_checks/solarwinds/solarwinds.py

```diff
@@ -0,0 +1,486 @@
+            # Is this NodeID present in the list of nodes?
+            found_node = next((sub for sub in nodes if sub.node_id == npm_node.get("NodeID")), None)
+            if not found_node:
+                # New node, create it and add it to the list
+                nodes.append(
+                    Node(
+                        npm_node.get("NodeID"),
+                        npm_node.get("Caption"),
+                        npm_node.get("IPAddress"),
+                        npm_node.get("NodeStatus"),
+                        npm_node.get(solarwinds_domain),
+                        npm_node.get("NodeCategory"),
+                        self.base_url + npm_node.get("NodeDetailsUrl"),
+                    )
+                )
+                found_node = nodes[-1]
+            # Is there an interface for this node?
+            interface_id = npm_node.get("InterfaceID")
+            if interface_id:
+                # Is this interface present in the list of interfaces on this node?
+                # For new nodes, this would not be necessary
+                found_interface = next((sub for sub in found_node.interfaces if sub.interface_id == interface_id), None)
+                if not found_interface:
+                    # New interface, create it and add it to the list
+                    found_node.append_interface(
+                        interface_id,
+                        npm_node.get("IfName"),
+                        npm_node.get("MAC"),
+                        npm_node.get("InterfaceStatus"),
+                        npm_node.get(solarwinds_domain),
+                        self.base_url + npm_node.get("InterfaceDetailsUrl"),
+                    )
+                    found_interface = found_node.interfaces[-1]
+                # Is there a connection to this interface?
+                dest_interface_id = npm_node.get("DestInterfaceID")
+                if dest_interface_id:
+                    # Is the destination node (of this interface) present in the solarwinds data?
+                    found_dest_node = next((sub for sub in npm_data if sub["InterfaceID"] == dest_interface_id), None)
+                    if found_dest_node:
+                        dest_node_name = found_dest_node.get("Caption")
+                        dest_interface_name = found_dest_node.get("IfName")
+                        dest_mac_address = npm_node.get("DestInterfaceMAC")
+                        found_interface.append_connection(dest_node_name, dest_interface_name, dest_mac_address)
+        return nodes
+
+    def check_udt_active(self):
+        self.log.info("Check for installed UDT module")
+        # Get a list of all installed Orion modules, check for (not expired) UDT
+        query_udt = self.swis.query(
+            "SELECT LicenseName, IsExpired FROM Orion.InstalledModule WHERE LicenseName = 'UDT' AND IsExpired = FALSE"
+        )
+        if query_udt["results"]:
+            self.log.info("UDT module is active")
```

Comment:
these are very minor level logging. so use debug to log it.

Reply:
done